ai/pathFinder.py

```diff
@@ -101,7 +101,7 @@ def path(self, start: Hex, end: Hex, exclude: Set[Hex]) -> List[Hex]:
                 break
 
             # Investigating each neighbor Hex of the current Hex.
-            for neighborHex in currentHex.neighbors():
+            for neighborHex in currentHex.range(0, speed+1):
```

Comment:
How is this supposed to work? A* need to examine only neighbor nodes. Later in the loop we construct node with `g` bigger by just one, this is not correct if we examine not only neighbor nodes

Reply:
I really didn't spend much time on it, but it should see all possible nodes that he can go, in range of speed 1 to maxspeed, later on it checks if it's moving through obstacle. That's better because we know that vehicle can go through vehicle.